packages/sdk/src/cross-chain-messenger.ts

```diff
@@ -367,6 +374,25 @@ export class CrossChainMessenger {
     }
   }
 
+  public async getWithdrawalsFromMessage(message: MessageLike) {
```

Comment:
how does this new function differ from `getMessagesByTranscation`? Can this existing function be used to derive the index passed to these messages.

At first glance, `getMessagesByTransaction` checks for `SentMessage` instead of the `MessagePassed` comparison here. Maybe i'm missing something but it looks okay to rely on `SentMessage` instead

Reply:
I think you are right but since I basically did 1to1 transforms I'm going to keep this the same and make it private to be safe